test/config/TestConfig.cpp

```diff
@@ -235,18 +256,284 @@ TEST_F(ConfigTestFixture, InvalidRootCaPathConfig)
     PlainConfig config;
     config.LoadFromJson(jsonView);
 
+    ASSERT_TRUE(config.Validate());
+    ASSERT_STREQ(rootCaPath.c_str(), config.rootCa->c_str());
+    ASSERT_STREQ("endpoint value", config.endpoint->c_str());
+    ASSERT_STREQ(filePath.c_str(), config.cert->c_str());
+    ASSERT_STREQ(filePath.c_str(), config.key->c_str());
+    ASSERT_STREQ("thing-name value", config.thingName->c_str());
+    ASSERT_FALSE(config.fleetProvisioning.enabled);
+    DefaultFeaturesEnabled(config);
+}
+
+/**
+ * Explicitly pass a valid root-ca path via CLI
+ * Expect Config.rootCa to equal root-ca path
+ */
+TEST_F(ConfigTestFixture, HappyCaseExplicitRootCaCli)
+{
+    CliArgs cliArgs = CliArgs{
+        {PlainConfig::CLI_ENDPOINT, "endpoint value"},
+        {PlainConfig::CLI_ROOT_CA, rootCaPath},
+        {PlainConfig::CLI_CERT, filePath},
+        {PlainConfig::CLI_KEY, filePath},
+        {PlainConfig::CLI_THING_NAME, "thing-name value"},
+    };
+
+    PlainConfig config;
+    config.LoadFromCliArgs(cliArgs);
+
+    ASSERT_TRUE(config.Validate());
+    ASSERT_STREQ(rootCaPath.c_str(), config.rootCa->c_str());
+    ASSERT_STREQ("endpoint value", config.endpoint->c_str());
+    ASSERT_STREQ(filePath.c_str(), config.cert->c_str());
+    ASSERT_STREQ(filePath.c_str(), config.key->c_str());
+    ASSERT_STREQ("thing-name value", config.thingName->c_str());
+    ASSERT_FALSE(config.fleetProvisioning.enabled);
+    DefaultFeaturesEnabled(config);
+}
+
+#if defined(MQTT_DISABLED)
```

Comment:
There is no macro named `MQTT_DISABLED` as a result these tests are never executed. You can confirm this by looking at the test output in CI for any of the build targets and observe that no test matching `ExplicitRootCaBadParentPermissionsConfig`. Before you spin another PR that replaces `MQTT_DISABLED` with `!defined(DISABLE_MQTT)` can you explain in the form of a comment why this definition becomes implicated in the test.  The reason I think this is important for exactly the kind of problem that you have stumbled onto -- preprocessor based includes can introduce hard to discover bugs.

Reply:
Discussed already but for posterity:

When built with MQTT_DISABLED Device Client was not validating the root-ca file permissions or parent directory permissions. So, this change was intended to not test the file permissions would fail validation if MQTT was disabled. However, realized now that we made root-ca optional and do not set a value unless it is valid then we can validate the file permissions if a value is present regardless of MQTT being disabled.